packages/vscode-extension/src/commonlib/m365Login.ts

```diff
@@ -308,7 +308,9 @@ export class M365Login extends BasicLogin implements M365TokenProvider {
       // If no account in cache file, we will ask to sign in directly.
       tokenRes = await this.signInWhenNoAccountInCache(tokenRequest, loginHint, true);
       VsCodeLogInstance.info(`Signed in with Microsoft 365 account: ${loginHint}`);
-    } else if (M365Login.codeFlowInstance.account.username === loginHint) {
+    } else if (
+      M365Login.codeFlowInstance.account.username.toLowerCase() === loginHint.toLowerCase()
```

Comment:
would username or loginHint be undefined?

Reply:
added check for loginHint.
We checked whether account is undefined or not in the if block above, so here the account is always AccountInfo type. And the type of AccountInfo in msal
export declare type AccountInfo = {
    homeAccountId: string;
    environment: string;
    tenantId: string;
    username: string;
    localAccountId: string;
    name?: string;
    idTokenClaims?: object;
};